otherlibs/stdune/lock_file.ml

```diff
@@ -1,27 +0,0 @@
-type t = Unix.file_descr
-
-let create_internal op path =
-  Path.mkdir_p (Path.parent_exn path);
-  let fd =
-    Unix.openfile (Path.to_string path) [ Unix.O_CREAT; Unix.O_RDWR ] 0o600
-  and pid = string_of_int (Unix.getpid ()) in
-  let pid_len = String.length pid in
-  (* fcntl would enable to downgrade the lock to a read lock so other process
-     can read the PID, but it is not bound in Unix *)
-  match
-    Unix.lockf fd op 0;
```

Comment:
This is dead code but `Unix.lockf` is a different API... however, https://0pointer.de/blog/projects/locking claims it's the same, 

and https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/lockf.3.html documents the same broken behavior (without any warning):

> File locks are released on first close by the locking process of any file descriptor for the file.